### ISSUE TITLE
Merge does not support None or non-dict types anymore

### ISSUE THREAD

**Issue by mattdodge** [dgilland/pydash#160]
This seems to have broken in pydash version 4.8.0 based on some bisecting.

With pydash 4.7.6
```
>>> import pydash as _
>>> print(_.merge(None, {'key': 'val'}))
None
```

With pydash 4.8.0
```
>>> import pydash as _
>>> print(_.merge(None, {'key': 'val'}))
AttributeError: 'NoneType' object has no attribute 'key'
```

Before, the merge method would just return the first argument if it wasn't a dictionary. This doesn't seem to be documented anywhere that I can find but we were sort of relying on it. Especially with None types.

Perhaps this is a "wontfix" but I wanted to bring it up as I didn't see any issues or any mention of it in the changelog.

**Comment by dgilland** [comment]
It wasn't intentional to make `pydash.merge(None, ...)` raise an exception so I don't see an issue with fixing that in 4.x.

However, looking at Lodash: `_.merge(null, {key: 'val'}) -> {key: 'val'}` and `_.merge(null) -> {}`. I'm not sure `pydash.merge(None, ...)` should be returning `None`, but pydash has never considered has `pydash.merge(None) == {}` so not sure if changing it for upcoming 5.x release is worthwhile at this point.

Just curious but what is your particular use-case for having `pydash.merge(None, ...)` return `None`?

**Comment by mattdodge** [comment]
I actually agree that essentially converting `None` to an empty dict would be the behavior that makes the most sense. The use case is more that the function call is resilient to `None` as an input rather than actually returning `None` as an output. It's obviously pretty easy to work around by wrapping in a simple conditional but it caused some test failures when I upgraded so I figured it was worth mentioning.
